### DISCUSSION TITLE
Why does GitHub API require admin rights to leave a comment on a PR?

### DISCUSSION THREAD

**Discussion by dahlia** [community/community#21556] [New to GitHub]
<p>I tried to create a comment using GitHub HTTP API and I got the following response with <em>403 Forbidden</em>:</p>
<pre><code>{  
 "message": "Must have admin rights to Repository.",  
 "documentation\_url": "https://developer.github.com/v3/issues/comments/#create-a-comment"  
}
</code></pre>
<p>The repository is not private but public, and the request was authenticated through a personal access token. (The response indeed contained <em>X-RateLimit-Limit: 5000</em> header, an evidence that implies the request was properly authenticated.)</p>
<p>What’s wrong with my use of GitHub API? Why does it requires admin rights to merely leave some comments?</p>

**Comment by dahlia** [comment, marked as answer]
<p>Solved — It was because the personal access token didn’t have not enough scope (<em>public_repo</em>).</p>

**Comment by nanisadw3** [comment]
This error usually happens because of one of two common reasons:
  
  1. **Targeting the wrong API endpoint:**
     - General comments on PRs/Issues only require **write (`repo` / `contents: write`)** access using:
       `POST /repos/{owner}/{repo}/issues/{issue_number}/comments`
     - If you are calling review endpoints or the repository requires branch protection reviews, permissions
can be escalated.
  2. **Fine-grained Personal Access Token (PAT) permissions:**
     - If using fine-grained tokens, you must explicitly grant **"Pull requests: Read & Write"** and
**"Issues: Read & Write"** on the specific target repository. Without write permission to those scopes, the
API rejects the request with a 403.
  3. **Organization Base Permissions:**
     - If the repository belongs to an organization, check the organization's **Member Privileges**. If
outside collaborators or members have default "Read" access, the token won't have permission to write comments
unless explicitly granted triage/write access.
  
  If this helped you troubleshoot the permission issue, feel free to mark this as the accepted answer!
